overlord.pl

```diff
@@ -342,13 +350,35 @@ sub finishCaching
     &waitFor("swapouts gone", sub { ! &squidHasSwapouts() });
 }
 
+sub waitActiveRequests
+{
+    my ($options) = @_;
+
+    my $requestPath = $options->{$onameRequestPath};
+    die("missing $onameRequestPath\n") unless defined $requestPath;
+    my $activeRequestsCount = $options->{$onameActiveRequestsCount};
+    die("missing $onameActiveRequestsCount\n") unless defined $activeRequestsCount;
+
+    my ($path, $count) = @_;
+    &waitFor("until all $activeRequestsCount requests become active", sub { &squidActiveRequests($requestPath, $activeRequestsCount) });
+}
+
 # whether Squid has StoreEntries in SWAPOUT_WRITING state
 sub squidHasSwapouts
 {
     my $mgrPage = &getCacheManagerResponse('openfd_objects')->{content};
     return $mgrPage =~ /SWAPOUT_WRITING/;
 }
 
+sub squidActiveRequests
+{
+    my $path = shift;
+    my $count = shift;
+    my $mgrPage = &getCacheManagerResponse('active_requests')->{content};
+    my @matches = $mgrPage =~ /$path/g;
```

Comment:
Let's be a bit more conservative in matching an externally-supplied string (rather than a well-known and fairly unique constant like `SWAPOUT_WRITING`):

```suggestion
    my @matches = $mgrPage =~ /^uri\s.*$path/mg;
```

Please note that I have not tested the above regex.

Reply:
Ok, I changed into `/^uri\s.*$path$/mg`